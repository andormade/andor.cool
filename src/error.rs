use std::fmt;
use std::io;

#[derive(Debug)]
pub enum Error {
    /// IO operation error
    Io(io::Error),
    /// Error that occurs during handlebars processing
    Handlebars(String),
}

impl fmt::Display for Error {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        match self {
            Error::Io(err) => write!(f, "IO error: {}", err),
            Error::Handlebars(msg) => write!(f, "Handlebars error: {}", msg),
        }
    }
}

impl std::error::Error for Error {
    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
        match self {
            Error::Io(err) => Some(err),
            _ => None,
        }
    }
}

impl From<io::Error> for Error {
    fn from(err: io::Error) -> Self {
        Error::Io(err)
    }
}

impl From<String> for Error {
    fn from(err: String) -> Self {
        Error::Handlebars(err)
    }
}

pub type Result<T> = std::result::Result<T, Error>; 