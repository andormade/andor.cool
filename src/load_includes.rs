use std::collections::HashMap;
use std::fs;
use std::path::Path;
use crate::error::MyError;

pub fn load_liquid_includes(dir_path: &str) -> Result<HashMap<String, String>, MyError> {
    let path = Path::new(dir_path);
    let mut templates = HashMap::new();

    for entry in fs::read_dir(path)? {
        let entry = entry?;
        let path = entry.path();
        if path.is_file() && path.extension().and_then(|ext| ext.to_str()) == Some("liquid") {
            if let Some(filename) = path.file_name().and_then(|name| name.to_str()) {
                let contents = fs::read_to_string(&path)?;
                templates.insert(filename.to_string(), contents);
            }
        }
    }

    Ok(templates)
}

#[cfg(test)]
mod tests {
    use super::*;
    use insta::assert_snapshot;
    use serde_json;
    use std::collections::BTreeMap;

    #[test]
    fn test_load_liquid_includes() {
        // Update test to handle Result
        let templates = load_liquid_includes("./_includes").unwrap_or_default();
        let sorted_templates: BTreeMap<_, _> = templates.into_iter().collect();
        let templates_json = serde_json::to_string_pretty(&sorted_templates).unwrap();
        // Use assert_snapshot! to compare the output against the stored snapshot
        assert_snapshot!(templates_json);
    }
}
