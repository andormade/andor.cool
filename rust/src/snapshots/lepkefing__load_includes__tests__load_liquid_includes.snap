---
source: src/load_includes.rs
expression: templates_json
---
{
  "candystore.liquid": "<a href=\"https://static.llllllllllll.com/andor/images/{{ file }}_original.jpg\"><img src=\"https://static.llllllllllll.com/andor/images/{{ file }}.jpg\" alt=\"{{ title }}\" title=\"{{ title }}\" loading=\"lazy\" /></a>\n",
  "emojiflag.liquid": "{% if location contains 'Australia' %}&#x1F1E6;&#x1F1FA;{% endif %}\n{% if location contains 'Austria' %}&#127462;&#127481;{% endif %}\n{% if location contains 'Belgium' %}&#127463;&#127466;{% endif %}\n{% if location contains 'Czech' %}&#127464;&#127487;{% endif %}\n{% if location contains 'Denmark' %}&#127465;&#127472;{% endif %}\n{% if location contains 'England' %}&#127468;&#127463;{% endif %}\n{% if location contains 'France' %}&#127467;&#127479;{% endif %}\n{% if location contains 'Germany' %}&#127465;&#127466;{% endif %}\n{% if location contains 'Hungary' %}&#127469;&#127482;{% endif %}\n{% if location contains 'India' %}&#127470;&#127475;{% endif %}\n{% if location contains 'Ireland' %}&#127470;&#127466;{% endif %}\n{% if location contains 'Italy' %}&#x1F1EE;&#x1F1F9;{% endif %}\n{% if location contains 'Japan' %}&#127471;&#127477;{% endif %}\n{% if location contains 'Latvia' %}&#127473;&#127483;{% endif %}\n{% if location contains 'Lithuania' %}&#127473;&#127481;{% endif %}\n{% if location contains 'Mexico' %}&#127474;&#127485;{% endif %}\n{% if location contains 'Netherlands' %}&#127475;&#127473;{% endif %}\n{% if location contains 'New Zealand' %}&#127475;&#127487;{% endif %}\n{% if location contains 'Norway' %}&#127475;&#127476;{% endif %}\n{% if location contains 'Romania' %}&#127479;&#127476;{% endif %}\n{% if location contains 'Russia' %}&#127479;&#127482;{% endif %}\n{% if location contains 'Scotland' %}󠁧󠁢󠁳󠁣󠁴&#127468;&#127463;{% endif %}\n{% if location contains 'South Korea' %}&#127472;&#127479;{% endif %}\n{% if location contains 'Sweden' %}&#127480;&#127466;{% endif %}\n{% if location contains 'Ukraine' %}&#127482;&#127462;{% endif %}\n{% if location contains 'United States' %}&#127482;&#127480;{% endif %}\n",
  "img.liquid": "<img src=\"{{ site.imagehost_url }}/{{ include.date | date: \"%Y\" }}/{{ include.img }}.jpg\" alt=\"{{ include.title }}\" title=\"{{ include.title }}\" />\n",
  "scope.liquid": "<img src=\"{{ site.imagehost_url }}/{{ include.date | date: \"%Y\" }}/{{ include.img }}.jpg\" alt=\"{{ include.title }}\" title=\"{{ include.title }}\" />\n",
  "ucare.liquid": "<a href=\"https://ucarecdn.com/{{ uuid }}/\"><img src=\"https://ucarecdn.com/{{ uuid }}/-/preview/2000x1000/\" alt=\"{{ title }}\" title=\"{{ title }}\" loading=\"lazy\" /></a>\n",
  "video.liquid": "<div class=\"video-container-container\">\n    <video width=\"100%\" controls preload=\"metadata\">\n        <source src=\"https://static.llllllllllll.com/andor/videos/{{id}}\" type=\"video/mp4\">\n    </video>\n</div>",
  "vimeo.liquid": "<div class=\"video-container-container\">\n\t<div class=\"video-container video-container-16-9\">\n\t\t<iframe src=\"https://player.vimeo.com/video/{{id}}\" frameborder=\"0\" allowfullscreen></iframe>\n\t</div>\n</div>",
  "youtube.liquid": "<div class=\"video-container video-container-16-9\">\n\t<iframe src=\"https://www.youtube-nocookie.com/embed/{{id}}\" title=\"YouTube video player\" frameborder=\"0\" allow=\"accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture\" allowfullscreen></iframe>\n</div>"
}
